core/src/main/java/com/radixdlt/RadixNode.java

```diff
@@ -0,0 +1,402 @@
+      // There's nothing on ledger and/or properties, so we can just use
+      // the network fixed genesis without any additional validation
+      startRadixNodeModule(fixedNetworkGenesis.get(), nodeBootStopwatch);
+    } else if (properties.get("genesis.use_olympia", BooleanUtils::parseBoolean).orElse(false)) {
+      // No genesis transaction known beforehand was found, but we can get it from Olympia...
+      startOlympiaGenesisModule(network, nodeBootStopwatch);
+    } else {
+      // TODO(post-babylon): remove Olympia ref from the message below
+      throw new RuntimeException(
+          "Radix node couldn't be initialized. No genesis transaction has been configured. Make"
+              + " sure that either `network.genesis_txn` or `network.genesis_file` is set or that"
+              + " you're using a well known network (`network.id`). If you're running an Olympia"
+              + " node consider using it as your genesis source (`genesis.use_olympia`). Refer to"
+              + " documentation for more details.");
+    }
+  }
+
+  private void startOlympiaGenesisModule(Network network, Stopwatch nodeBootStopwatch) {
+    final var genesisFromOlympiaInjector =
+        Guice.createInjector(new GenesisFromOlympiaNodeModule(properties, network));
+    this.maybeGenesisFromOlympiaInjector = Optional.of(genesisFromOlympiaInjector);
+
+    log.info(
+        "Olympia-based genesis was configured ({}). Using core API URL of {}",
+        network.getLogicalName(),
+        genesisFromOlympiaInjector.getInstance(OlympiaGenesisConfig.class).nodeCoreApiUrl());
+
+    final var systemApi = genesisFromOlympiaInjector.getInstance(SystemApi.class);
+    systemApi.start();
+
+    final var olympiaGenesisService =
+        genesisFromOlympiaInjector.getInstance(OlympiaGenesisService.class);
+
+    olympiaGenesisService.start(
+        genesisData ->
+            proceedWithGenesisFromOlympia(
+                nodeBootStopwatch, genesisFromOlympiaInjector, genesisData),
+        ex -> {
+          log.warn(
+              "Radix node couldn't be initialized. The Olympia-based genesis was configured but"
+                  + " an error occurred.",
+              ex);
+          this.shutdown();
+        });
+  }
+
+  private void proceedWithGenesisFromOlympia(
+      Stopwatch nodeBootStopwatch, Injector genesisFromOlympiaInjector, GenesisData genesisData) {
+    log.info(
+        "Genesis data has been successfully received from the Olympia node ({} accounts,"
+            + " {} validators). Initializing the Babylon node...",
+        genesisData.accountXrdAllocations().size(),
+        genesisData.validatorSetAndStakeOwners().size());
+    cleanupOlympiaGenesisModule(genesisFromOlympiaInjector);
+    final var genesisTxn = genesisData.toGenesisTransaction(GenesisConfig.babylonDefault());
+    this.startRadixNodeModule(genesisTxn, nodeBootStopwatch);
+  }
+
+  private void cleanupOlympiaGenesisModule(Injector genesisFromOlympiaInjector) {
```

Comment:
`OlympiaGenesisBootstrapper` sounds like a good idea, I'll give it a try :+1: 

The latter suggestion seems a bit leaky, abstraction wise. I'd prefer RadixNode to be a self contained component, independent of the application.

Reply:
> leaky

That's true (accessing the `.metrics()` from outside), but you could say that "measure your own boot up time" is also leaky (typically, the owner [or decorator] layer instruments the underlying one: starts the timer, calls a method, observes the duration).

Ofc that's only conceptual, and from practical PoV I am ok with RadixNode measuring itself 👍 

I wondered why this cannot be a bit more elegant (i.e. not pass the `nodeBootStopwatch` all the way through the call stack), and I realized that this `.start()` method is actually async! (due to `OlympiaGenesisService`'s callbacks). In that case, shouldn't it rather return a `Future<Void>`? (in case the caller wanted to block until it really started, _e.g. to make a time measurement :p or to log "alright I started"_)

_(a conversion from callbacks to `Future` is trivial - like, call `CompletableFuture.complete(x)` from the callback, and it really makes things more obvious [think about what made the oldschool javascript programming a hell])_

Reply:
I guess it depends how we define "starting a node" :) In the end the Olympia bootstrapping mechanism is a part of the node, so the node is technically _started_ when it's running, but _uninitialized_(?). I've considered returning a Future from `start` but then realized it wouldn't serve any real purpose :)

Reply:
I went with a separate `RadixNodeBootstrapper` - let me know what you think.